consai_visualizer/src/consai_visualizer/visualizer.py

```diff
@@ -120,29 +121,166 @@ def __init__(self, context):
         self.latest_battery_voltage = [0] * 16
         self.latest_kicker_voltage = [0] * 16
 
+        # 設定保存ファイル（installディレクトリからsrc/consai_ros2に移動）
+        pkg_name = "consai_visualizer"
+        _, package_path = get_resource("packages", pkg_name)
+        # /install/consai_visualizer -> /src/consai_ros2
+        base_path = package_path.replace("install/consai_visualizer", "src/consai_ros2")
+        self._custom_settings_file = os.path.join(base_path, "consai_visualizer/consai_visualizer_settings.json")
+        self._default_settings_file = os.path.join(
+            base_path, "consai_visualizer/default_consai_visualizer_settings.json"
+        )
+
+        # 起動時の復元処理中は保存を無効にするフラグ
+        self._is_loading_settings = False
+        # 初期化完了フラグ（新しいレイヤー追加時の保存を防ぐ）
+        self._initialization_complete = False
+        # 復元待ちの設定データ
+        self._pending_settings = None
+
     def save_settings(self, plugin_settings, instance_settings):
-        # UIを終了するときに実行される関数
+        # RQT標準の保存（何もしない）
+        pass
+
+    def _save_settings_to_file(self):
+        # 独自設定ファイルに保存
+        try:
+            active_layers = self._extract_active_layers()
+            if not active_layers:
+                self._logger.info("No active layers to save")
+                return
+
+            # レイヤー情報を準備
+            layer_data = [{"layer": layer, "sub_layer": sub_layer} for layer, sub_layer in active_layers]
+
+            settings = {"active_layers": layer_data, "timestamp": time.time()}
+
+            # ディレクトリが存在することを確認
+            settings_dir = os.path.dirname(self._custom_settings_file)
+            if not os.path.exists(settings_dir):
+                os.makedirs(settings_dir)
+                self._logger.info(f"Created directory: {settings_dir}")
+
+            # JSONファイルに保存
+            self._logger.info(f"Attempting to save to: {self._custom_settings_file}")
+            with open(self._custom_settings_file, "w") as f:
+                json.dump(settings, f, indent=2)
+
+            # ファイルが実際に作成されたか確認
+            if os.path.exists(self._custom_settings_file):
+                self._logger.info(f"Successfully saved {len(layer_data)} layers to {self._custom_settings_file}")
```

Comment:
consai起動時にここのログ表示がたくさん出るから、debugにしてもいいかも
```suggestion
                self._logger.debug(f"Successfully saved {len(layer_data)} layers to {self._custom_settings_file}")
```